[alyssaongyx] iP

alyssaongyx opened the pull request:
## Ally
> Ally is your personalised chatbot so that you no longer struggle with your workload!

### Why you should use Ally
Ally is...
- text-based
- easy to learn and use 
- ~~FAST~~ **_SUPER_**  FAST to use

### What features does Ally have?
- [x] Managing tasks
- [ ] Managing deadlines(coming soon!)
- [ ] Reminders(coming soon!)

### Steps to get Ally
All you need to do is,
1. download it from [here](https://github.com/alyssaongyx/ip)
2. run it! :ghost:

And it is **FREE**! So what are you waiting for?

### Resource
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
``` ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

alyssaongyx committed:
Increment to Level-0: Rename, Greet, Exit

alyssaongyx committed:
Increment to Level-0

alyssaongyx committed:
Increment to Level-1

alyssaongyx committed:
New update

alyssaongyx committed:
Updated

alyssaongyx committed:
Increment to Level 2

alyssaongyx committed:
Increment to Level 3

alyssaongyx committed:
Level 4

alyssaongyx committed:
A-TextUiTesting initial commit

alyssaongyx committed:
Increment to Level-5

alyssaongyx committed:
Increment to Level-6

alyssaongyx committed:
Updated the print statements

alyssaongyx committed:
Added comments

alyssaongyx committed:
level-7

alyssaongyx committed:
Update Level-7

alyssaongyx committed:
Update deadline file

alyssaongyx committed:
More OOP

alyssaongyx committed:
Create new packages

alyssaongyx committed:
Stage saved.txt

alyssaongyx committed:
added javadocs

alyssaongyx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

* origin/add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

alyssaongyx committed:
Set up gradle

alyssaongyx committed:
no message

alyssaongyx committed:
Created JUnit Tests

alyssaongyx committed:
Updated build gradle to generate Jar file

alyssaongyx committed:
Added javadocs to the public methods

alyssaongyx committed:
Update to comply with coding standard

alyssaongyx committed:
Created FindCommand

alyssaongyx committed:
Level-9

lamchenghou commented on src/main/java/Ally/Ally.java:
```diff
@@ -0,0 +1,58 @@
+package Ally;
```
I believe package names should all be in lowercase. Good job on having the correct package structure!

lamchenghou commented on src/main/java/Ally/Ally.java:
```diff
@@ -0,0 +1,58 @@
+     * Catches AllyException.
+     *
+     * @param filePath
+     * @throws AllyException
```
Perhaps you can give descriptions to your JavaDocs parameters and exceptions.

lamchenghou commented on src/main/java/Ally/Ally.java:
```diff
@@ -0,0 +1,58 @@
+            } catch (AllyException e) {
+                ui.showError(e.getMessage());
+            } finally {
+                Ui.showLine();
```
I like that you have a single level of abstraction here.

lamchenghou commented on src/main/java/Ally/Commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,48 @@
+package Ally.Commands;
+
+import Ally.*;
```
Is it better to list imported classes explicitly?

lamchenghou commented on src/main/java/Ally/Parser.java:
```diff
@@ -0,0 +1,43 @@
+     * @return
+     * @throws AllyException
+     */
+    public static Commands parse(String ipt) throws AllyException {
```
```suggestion
    public static Commands parse(String input) throws AllyException {
```
Perhaps you can use standard words to improve clarity.

lamchenghou commented on src/main/java/Ally/Parser.java:
```diff
@@ -0,0 +1,43 @@
+            return new UnmarkCommand(Integer.parseInt(split[1]) - 1);
+        } else if (split[0].equals("delete")) {
+            return new DeleteCommand(Integer.parseInt(split[1]) - 1);
+        } else if (split[0].equals(("find"))) {
```
Nit: As`split[0]` is used across multiple statements, you can assign it to a noun before using it, perhaps `String potentialCommand = split[0];`.

I also think that the naming of variables `splits1` and `splits2` can be more descriptive, perhaps `firstSubstring` and `secondSubstring`. 

Good job on the parsing! :)

lamchenghou commented on src/main/java/Ally/Tasks/Event.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String formatFile() {
+        return "E" + " | " + (isDone ? "1" : "0") + " | " + description + " | "
```
A little tip: I find `String::format` very useful when I want to put Strings and variables together. For instance, `String.format("E | %s | %s | %s | %s", isDone ? "1" : "0", description, from, to);`.

lamchenghou commented on src/test/java/Ally/Tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,68 @@
+public class DeadlineTest {
+
+    @Test
+    public void getStatusIcon_markAsDone_success(){
```
Good job on the test case naming 😃.

lamchenghou reviewed (commented):
LGTM 👍, overall a readable and clear codebase, with proper abstractions and decent naming. I enjoyed reading through your code. There weren't any huge code quality issues, and SLAP/KISS principles were applied .

There are some code quality issues and nits to fix.

alyssaongyx committed:
Correct checkstyle errors.

techjay-c commented on src/main/java/ally/Parser.java:
```diff
@@ -0,0 +1,55 @@
+     * @return commands
+     * @throws AllyException Error Message
+     */
+    public static Commands parse(String ipt) throws AllyException {
```
very neat categorisation of the commands. keep it up

techjay-c commented on src/main/java/ally/commands/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public void run(AllyList allyList, Ui ui, Storage storage) {
+        ArrayList<Task> matchTasks = new ArrayList<>(100);
```
maybe you want to consider not declaring a fixed size ArrayList

techjay-c reviewed (commented):
Peer Review Comments

alyssaongyx committed:
Set up GUI

alyssaongyx committed:
Add assertions

Commands have run method which contains allylist, ui and storage attributes.

The attributes can be empty. However, commands like delete command could still occur.

Assertions causes exceptions when this issue arises.

alyssaongyx committed:
Merge pull request #1 from alyssaongyx/branch-A-Assertions

Add assertions

alyssaongyx committed:
Update quality of code.

Remove unnecessary large blocks of comments.

Large blocks of comments would make the code untidy.

alyssaongyx committed:
Merge pull request #2 from alyssaongyx/branch-A-CodeQuality

Update quality of code

alyssaongyx committed:
Add GitHub actions

Add the gradle.yml file.

alyssaongyx committed:
Removed the gradle file due to some errors

alyssaongyx committed:
Create gradle.yml

alyssaongyx committed:
Update gradle.yml

alyssaongyx committed:
Update to remove checkstyle errors

alyssaongyx committed:
Merge branch 'master' of https://github.com/alyssaongyx/ip

* 'master' of https://github.com/alyssaongyx/ip:
  Update gradle.yml

alyssaongyx committed:
Add Sort command for deadline tasks

alyssaongyx committed:
Merge pull request #4 from alyssaongyx/branch-BCD-Extension

Add Sort command for deadline tasks

alyssaongyx committed:
Add Help Command to show the list of formats of the commands for a new user

alyssaongyx committed:
Add Todo empty argument exception

alyssaongyx committed:
Add a screenshot in the docs folder

alyssaongyx committed:
Update README.md

alyssaongyx committed:
Update README

alyssaongyx committed:
Reformat README
